### PR TITLE
Add doctest examples and update spec for Enum.zip

### PR DESCRIPTION
**Pull request by solar05**

### CODE REVIEW

**Comment by josevalim** [comment]
The current spec already includes empty lists and we already have an example for mismatched lists, so I think se can skip this one. :)

File: lib/elixir/lib/enum.ex [hunk 1]
```diff
@@ -3903,8 +3903,11 @@ defmodule Enum do
       iex> Enum.zip([1, 2, 3, 4, 5], [:a, :b, :c])
       [{1, :a}, {2, :b}, {3, :c}]
 
+      iex> Enum.zip([], [:a, :b, :c])
+      []
+
   """
-  @spec zip(t, t) :: [{any, any}]
+  @spec zip(t, t) :: [{any, any}] | []
```
Comment (sabiwara) [inline lib/elixir/lib/enum.ex:3910]: `[t]` is like `list(t)`, not `nonempty_list(t)`, so this is unnecessary
```suggestion
  @spec zip(t, t) :: [{any, any}]
```

File: lib/elixir/lib/enum.ex [hunk 2]
```diff
@@ -3927,6 +3930,9 @@ defmodule Enum do
       iex> Enum.zip([[1, 2, 3, 4, 5], [:a, :b, :c]])
       [{1, :a}, {2, :b}, {3, :c}]
 
+      iex> Enum.zip([[], [:a, :b, :c]])
+      []
```
Comment (sabiwara) [inline lib/elixir/lib/enum.ex:3934]: 👍

File: lib/elixir/lib/enum.ex (hunk 1 above)
Comment (solar05) [reply lib/elixir/lib/enum.ex]: Missed that, thanks!

**Pull request closed without merging**